qiskit/quantum_info/operators/operator.py

```diff
@@ -341,10 +347,10 @@ def tensor(self, other):
         """
         if not isinstance(other, Operator):
             other = Operator(other)
-        input_dims = other.input_dims() + self.input_dims()
-        output_dims = other.output_dims() + self.output_dims()
-        data = np.kron(self._data, other._data)
-        return Operator(data, input_dims, output_dims)
+        ret = copy.copy(self)
```

Comment:
these 3 lines appear exactly also in `statevector` and `densitymatrix` - is this code duplication needed?
similar comment is relevant also for the `expand` function.

Reply:
I think this is ok. Since base operator doesn't assume anything about how data is stored in subclasses, those subclasses which have same sort of array structure will have some small amount of duplication. It will get improved a little bit more in #5617 and follow up PRs.